packages/modules/user/src/services/user-module.ts

```diff
@@ -62,21 +62,42 @@ export default class UserModuleService<
 
     this.baseRepository_ = baseRepository
     this.userService_ = userService
-    this.inviteService_ = inviteService.withModuleOptions(
-      this.moduleDeclaration
-    )
+    this.inviteService_ = inviteService
+    this.config = {
+      jwtSecret: moduleDeclaration["jwt_secret"],
+      expiresIn:
+        parseInt(moduleDeclaration["valid_duration"]) ||
+        DEFAULT_VALID_INVITE_DURATION,
+    }
   }
 
   @InjectTransactionManager("baseRepository_")
   async validateInviteToken(
     token: string,
     @MedusaContext() sharedContext: Context = {}
   ): Promise<UserTypes.InviteDTO> {
-    const invite = await this.inviteService_.validateInviteToken(
-      token,
+    const jwtSecret = this.moduleDeclaration["jwt_secret"]
+    if (!jwtSecret) {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_DATA,
+        "No jwt_secret was provided in the UserModule's options. Please add one."
+      )
+    }
```

Comment:
thought: We throw this error in two places in critical flows. Would it make sense to check for this in the service constructor instead and throw at registration-time instead?

So when set on `this.config`

Reply:
Yes I think it would be better to not get the error too later

Reply:
Yes definitely, let me clean it up.